File: api/idlix.js
const { dec } = require('./CryptoJsAes');
const fetch = require('node-fetch');
const CryptoJsAes = require('./CryptoJsAes');

class Idlix {
    constructor(videoId) {
        this.BASE_WEB_URL = 'https://tv4.idlix.asia/';
        this.video_id = videoId;
        this.embed_url = null;
    }

    async getEmbedUrl() {
        if (!this.video_id) {
            return {
                status: false,
                message: 'Video ID is required'
            };
        }

        try {
            const response = await fetch(`${this.BASE_WEB_URL}wp-admin/admin-ajax.php`, {
                method: 'POST',
                headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
                body: new URLSearchParams({
                    action: 'doo_player_ajax',
                    post: this.video_id,
                    nume: '1',
                    type: 'movie'
                })
            });

            const jsonResponse = await response.json();

            if (response.status === 200 && jsonResponse.embed_url) {
                this.embed_url = CryptoJsAes.decrypt(
                    jsonResponse.embed_url,
                    dec(jsonResponse.key, JSON.parse(jsonResponse.embed_url).m)
                );

                return {
                    status: true,
                    embed_url: this.embed_url
                };
            } else {
                return {
                    status: false,
                    message: 'Failed to get embed URL'
                };
            }
        } catch (error) {
            return {
                status: false,
                message: error.toString()
            };
        }
    }
}

// Fungsi utama untuk menjalankan logika
(async () => {
    const videoId = '124474'; // ID Video yang diinginkan

    const idlix = new Idlix(videoId);
    const result = await idlix.getEmbedUrl();
    console.log(result); // Hasil embed URL atau error
})();

File: idlix.py
import requests
import json
from crypto_helper import CryptoJsAes, dec

class Idlix:
    BASE_WEB_URL = "https://tv10.idlixku.com/"  # Ubah dengan URL dasar yang sesuai
    video_id = None
    embed_url = None

    def __init__(self, video_id):
        self.video_id = video_id

    def get_embed_url(self):
        if not self.video_id:
            return {
                'status': False,
                'message': 'Video ID is required'
            }
        try:
            request = requests.post(
                url=self.BASE_WEB_URL + "wp-admin/admin-ajax.php",
                data={
                    "action": "doo_player_ajax",
                    "post": self.video_id,
                    "nume": "1",
                    "type": "movie",
                }
            )
            if request.status_code == 200 and request.json().get('embed_url'):
                self.embed_url = CryptoJsAes.decrypt(
                    request.json().get('embed_url'),
                    dec(
                        request.json().get('key'),
                        json.loads(request.json().get('embed_url')).get('m')
                    )
                )
                return {
                    'status': True,
                    'embed_url': self.embed_url
                }
            else:
                return {
                    'status': False,
                    'message': 'Failed to get embed URL'
                }
        except Exception as error_get_embed_url:
            return {
                'status': False,
                'message': str(error_get_embed_url)
            }
